Code/Editor/NewLevelDialog.cpp

```diff
@@ -206,7 +207,27 @@ void CNewLevelDialog::OnLevelNameChange()
     UpdateData(true);
 
     // QRegExpValidator means the string will always be valid as long as it's not empty:
-    const bool valid = !m_level.isEmpty() && ValidateLevel();
+    bool valid = !m_level.isEmpty() && ValidateLevel();
+    if (valid)
+    {
+        QDir levelDir(QString("%1/%2/").arg(m_levelFolders, m_level));
+        QString strLevelPath = levelDir.absoluteFilePath(m_level + EditorUtils::LevelFile::GetDefaultFileExtension());
+        int levelMaxLength = (AZ::IO::MaxPathLength - m_levelFolders.length() - QString(EditorUtils::LevelFile::GetDefaultFileExtension()).length() - 2) / 2;
```

Comment:
What is the calculation that is occurring here?
The subtraction by 2 is a magic value with no explanation. Also what is the division by 2 for?